plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -27,65 +27,140 @@
+        Optional<String> vaultAppRole = getVariable(CASC_VAULT_APPROLE, prop);
+        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
+        Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
+        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
+        Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
+                .map(Optional::of)
+                .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
+
+        // Check mandatory variables are set
+        if(!vaultUrl.isPresent()) {
+            LOGGER.log(Level.WARNING, "Mandatory variable {0} not set. Cannot fetch from vault.", CASC_VAULT_URL);
+            return;
+        }
+        if(!vaultPaths.isPresent()) {
+            LOGGER.log(Level.WARNING, "Mandatory variable {0} not set. Cannot fetch from vault.", CASC_VAULT_PATHS);
+            return;
+        }
+
+        // configure vault client
+        Vault vault = null;
+        VaultConfig vaultConfig = null;
+        try {
+            vaultConfig = new VaultConfig().address(vaultUrl.get());
+            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl.get());
+            if (vaultNamespace.isPresent()) {
+                vaultConfig.nameSpace(vaultNamespace.get());
             }
+            if (vaultEngineVersion.isPresent()) {
+                vaultConfig.engineVersion(Integer.parseInt(vaultEngineVersion.get()));
+            }
+            vaultConfig = vaultConfig.build();
+            vault = new Vault(vaultConfig);
+        } catch (VaultException e) {
+            LOGGER.log(Level.WARNING, "Could not configure vault connection", e);
         }
-        String vaultPw = getVariable("CASC_VAULT_PW", prop);
-        String vaultUsr = getVariable("CASC_VAULT_USER", prop);
-        String vaultPth = getVariable("CASC_VAULT_PATH", prop);
-        String vaultUrl = getVariable("CASC_VAULT_URL", prop);
-        String vaultMount = getVariable("CASC_VAULT_MOUNT", prop);
-        String vaultToken = getVariable("CASC_VAULT_TOKEN", prop);
-        String vaultAppRole = getVariable("CASC_VAULT_APPROLE", prop);
-        String vaultAppRoleSecret = getVariable("CASC_VAULT_APPROLE_SECRET", prop);
-        String vaultNamespace = getVariable("CASC_VAULT_NAMESPACE", prop);
-        String vaultEngineVersion = getVariable("CASC_VAULT_ENGINE_VERSION", prop);
-
-        if(((vaultPw != null && vaultUsr != null) || 
-            vaultToken != null || 
-            (vaultAppRole != null && vaultAppRoleSecret != null)) && vaultPth != null && vaultUrl != null) {
-            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl);
+
+        Optional<String> authToken = Optional.empty();
+
+        // attempt token login
+        if (vaultToken.isPresent() && !authToken.isPresent()) {
+            authToken = Optional.of(vaultToken.get());
```

Comment:
We seem to have lost the
```
LOGGER.log(Level.FINE, "Using supplied token to access Vault");
```

Reply:
Yes you are right. That one got lost. Added it again :+1: